src/main/java/com/marklogic/spark/writer/ArbitraryRowConverter.java

```diff
@@ -25,33 +27,46 @@
  */
 class ArbitraryRowConverter implements RowConverter {
 
-    private static ObjectMapper objectMapper = new ObjectMapper();
+    private final ObjectMapper objectMapper;
 
     private final StructType schema;
     private final String uriTemplate;
     private final String jsonRootName;
+    private final String xmlRootName;
+    private final String xmlNamespace;
 
     ArbitraryRowConverter(WriteContext writeContext) {
         this.schema = writeContext.getSchema();
         this.uriTemplate = writeContext.getStringOption(Options.WRITE_URI_TEMPLATE);
         this.jsonRootName = writeContext.getStringOption(Options.WRITE_JSON_ROOT_NAME);
+        this.xmlRootName = writeContext.getStringOption(Options.WRITE_XML_ROOT_NAME);
+        this.xmlNamespace = writeContext.getStringOption(Options.WRITE_XML_NAMESPACE);
+        this.objectMapper = new ObjectMapper();
     }
 
     @Override
     public Optional<DocBuilder.DocumentInputs> convertRow(InternalRow row) {
         final String json = convertRowToJSONString(row);
-        AbstractWriteHandle contentHandle = new StringHandle(json).withFormat(Format.JSON);
-        ObjectNode columnValues = null;
+        AbstractWriteHandle contentHandle = this.xmlRootName != null ?
+            new StringHandle(convertJsonToXml(json)).withFormat(Format.JSON) :
```

Comment:
This confuses me. If you're converting it to XML, why is Format still JSON?

Reply:
That's a bug - though it doesn't produce a failure because the "Format" is ignored in favor of the URI extension........... making me wonder if Format should even be set. I'll change it to XML, but I may just remove it.

Reply:
Actually, Format is worth keeping if the user chooses a URI extension that MarkLogic doesn't recognize. Would be very odd, but legit.